src/Gzero/Core/Http/Middleware/MultiLanguage.php

```diff
@@ -47,21 +48,37 @@ public function handle($request, Closure $next)
         /** @var LanguageService $languageService */
         $languageService = resolve(LanguageService::class);
         $languages       = $languageService->getAllEnabled();
-        $language        = $languages->first(function ($language) use ($request) {
-            return $language->code === $request->segment(1);
-        });
+        $language        = null;
 
-        if (!empty($language)) {
-            app()->setLocale($language->code);
+        if (str_is('api.*', $request->getHost())) {
+            $language = $languages->first(function ($language) use ($request) {
+                return $language->code === $request->header('Accept-Language');
+            });
         } else {
-            $language = ServiceProvider::setDefaultLocale();
+            $language = $languages->first(function ($language) use ($request) {
+                return $language->code === $request->segment(1);
+            });
         }
 
+        if (empty($language)) {
+            $language = $languageService->getDefault();
+        }
+
+        if (empty($language)) {
+            throw new DomainException('No default language found');
+        }
+
+        app()->setLocale($language->code);
+
         // We need it to make carbon's diffForHumans work correctly
         Carbon::setLocale(app()->getLocale());
```

Comment:
I would rewrite it for better simplicity to this
```
Carbon::setLocale($language->code)
```
However it is a matter of a taste